libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -604,9 +604,10 @@ class MapboxNavigation(
     private fun createInternalOffRouteObserver() = object : OffRouteObserver {
         override fun onOffRouteStateChanged(offRoute: Boolean) {
             if (offRoute) {
+                if (rerouteController?.state == RerouteState.FetchingRoute) {
```

Comment:
I imagine this logic should be moved into RerouteController

Reply:
Agreed :+1: